addons/multiplegap/src/presenter.js

```diff
@@ -1727,26 +1744,101 @@ function Addonmultiplegap_create(){
+                        answerHTML = prepareCheckedAnswerToPrint (answerHTML, i, ...studentAnswers);
+                    }
+                } else {
+                    answerHTML = prepareBasicPrint (answerHTML, ...studentAnswers);
+                }
             } else {
-                answerHTML = answerArray.join("</br>");
+                if (showAnswers) {
+                    answerHTML = prepareBasicPrint (answerHTML, ...correctAnswers);
+                }
             }
             $wrapper.html(answerHTML);
         }
 
         $view.append($wrapper);
 
-
         return $view[0].outerHTML;
     };
 
+    function prepareBasicPrint (answerHTML, ...answers) {
+        if (presenter.configuration.orientation === presenter.ORIENTATIONS.HORIZONTAL) {
+            for (var i = 0; i < answers.length; i++) {
+                answerHTML = answerHTML + answers[i].value;
+                if (i < answers.length - 1) {
+                    answerHTML = answerHTML + ", ";
+                }
+            }
+        } else {
+            for (var i = 0; i < answers.length; i++) {
+                answerHTML = answerHTML + answers[i].value;
+                if (i < answers.length - 1) {
+                    answerHTML = answerHTML + "</br>";
+                }
+            }
+        }
+        return answerHTML;
+    }
+
+    function parseStudentAnswers (state) {
+        var studentAnswers = [];
+        for (var i = 0; i < state.placeholders.length; i++){
+            var splitItemAnswerID = state.placeholders[i].item.split("-");
+            if (splitItemAnswerID.length === 2 && !isNaN(splitItemAnswerID[1])){
+                var answerAddonID = splitItemAnswerID[0];
+                var answerItemIndex = splitItemAnswerID[1] - 1;
+                var answer = {
+                    value : state.placeholders[i].value,
+                    addonID : answerAddonID,
+                    index : answerItemIndex,
+                    isCorrect : false
+                };
+                studentAnswers.push(answer);
+            }
+        }
+        return studentAnswers;
+    }
+
+    function prepareCheckedAnswerToPrint (answerHTML, i, ...answers){
+        if (answers[i].isCorrect) {
+            if (i < answers.length - 1) {
+                answers[i].value = '<div><span class=answerSpan>' + answers[i].value + '</span><span class="correctAnswerSpan"></span>, </div>';
+            } else {
+                answers[i].value = '<div><span class=answerSpan>' + answers[i].value + '</span><span class="correctAnswerSpan"></span></div>';
+            }
+        } else {
+            if (i < answers.length - 1) {
+                answers[i].value = '<div><span class=answerSpan>' + answers[i].value + '</span><span class="inCorrectAnswerSpan"></span>, </div>';
+            } else {
+                answers[i].value = '<div><span class=answerSpan>' + answers[i].value + '</span><span class="inCorrectAnswerSpan"></span></div>';
```

Comment:
przeniósłbym incorrectAnswerSpan (zmień z inCorrect na incorrect) i correctAnswerSpan do "głównego" spana i zmienił w cssach z before na after. Efekt będzie ten sam, ale nie będziesz miał nadmiarowego spana. Do tego wywal te div'y i br'y - chyba się nie dogadaliśmy tutaj o co mi chodziło, więc może to pomińmy. Tak jak to teraz jest zrobione, to tryb showAnswers+state wygląda inaczej niż wszystkie inne.